lib/lintSource.js

```diff
@@ -65,13 +67,30 @@ module.exports = async function lintSource(stylelint, options = {}) {
 
 	const config = configForFile.config;
 	const existingPostcssResult = options.existingPostcssResult;
+	const fileCache = options.fileCache;
+
+	if (options.cache && fileCache) {
+		const stylelintVersion = pkg.version;
+		const hashOfConfig = hash(`${stylelintVersion}_${JSON.stringify(config || {})}`);
```

Comment:
[ask] I worry we calculate the hash of a config file each time a single file is analyzedf. What do you think?

Reply:
Thanks. It is true, this it is not a good idea to calculate each file.
I think I can improve it thanks to your suggested comments!

Reply:
I Added a check to `fileCache` has already `hashOfConfig`. [`ca2fb53`](https://github.com/stylelint/stylelint/pull/6356/commits/ca2fb534ba93e2dba40092945e6f56758c99a97e#diff-37813c24c3fa3c6093015b61419f0bc18983cc2bbaddd56a732576bdb53f9023)